Tests/DatadogTests/Datadog/Core/AutoInstrumentation/URLSessionSwizzlerTests.swift

```diff
@@ -9,10 +9,13 @@ import XCTest
 
 class URLSessionSwizzlerTests: XCTestCase {
     /// URL.mockAny() is a valid URL so that it loads actual resource and exceeds expectation timeouts
-    let mockURL = URL(string: "https://foo.bar")!
-    let mockURLRequest = URLRequest(url: URL(string: "https://foo.bar")!)
-    let modifiedURLRequest = URLRequest(url: URL(string: "https://bar.foo")!)
-    let secondModifiedURLRequest = URLRequest(url: URL(string: "https://bar.foo.bar")!)
+    /// We use unsupported URLs so that the task goes through its URLProtocol chain and **immediately** returns with "unsupported URL" error
+    let mockURL = URL(string: "foo://example.com")!
```

Comment:
Is this a documented behaviour of `URLSession` or our observation?

Reply:
going through `URLProtocol` chain is documented in `URLProtocol`, the rest is observation
but "immediately" may be misunderstood: i mean "without making a connection attempt"

Reply:
Ok, sounds good enough. Just a note - if using `ServerMock` which mocks the `URLProtocol` we wouldn't have to explain anything in the comment 😉.

Reply:
yes, `URLProtocol` way could be nice but it doesn't work with `URLSession.shared`

Reply:
Yup, but we don't need to unit-test `URLSession.shared`, any other session created within the app will be a subject for interception (through swizzling), so `server.session` may also work. We cover `URLSession.shared` in integration tests, so it's safe at any front.

Anyway, that's just an idea, doesn't change the fact it works. Good to go for me 👌.

Reply:
sure, we could do that 👌 

the idea of testing `URLSession.shared/(configuration:)/(config:delegate:queue:)` came when i realized those methods return private `URLSession` subclasses, they may return different types one day (i admit this is a bit of over-caution ⚠️ )
as you said, we've been safe since we covered `.shared` in integration, now we can change if needed 👍